src/Engine/ProtoCore/Utils/CompilerUtils.cs

```diff
@@ -336,6 +336,8 @@ private static bool CompileCodeBlockAST(Core core, ParseParam parseParams, IDict
                     reWrittenNodes = MigrationRewriter.MigrateMethodNames(reWrittenNodes, priorNames, core.BuildStatus.LogDeprecatedMethodWarning);
                 }
 
+                ParserUtils.AssemblyDetector.GetAssemblyFromCodeBlock(core.ClassTable, reWrittenNodes);
```

Comment:
This is the main method to call for a CBN. At the moment this is being called from within DynamoCore itself where it has access to the AST's parsed from the CBN (namely, `rewrittenNodes`) as well as the VM classtable.